scipy/special/_logsumexp.py

```diff
@@ -93,43 +96,49 @@ def logsumexp(a, axis=None, b=None, keepdims=False, return_sign=False):
     1.6094379124341005, 1.6094379124341005
 
     """
-    a = _asarray_validated(a, check_finite=False)
```

Comment:
I don't think we can quite replace `_asarray_validated` with the checks in `array_namespace`. Specifically, the `sparse_ok=False` part. I do think that it is worth incorporating `_asarray_validated` into `array_namespace` though, which should just mean adding a `sparse_ok` kwarg which gets passed to `compliance_scipy`, I think.

Reply:
Adding a sparse check to `array_namespace` sounds like a good Idea. If the goal of those checks is to exclude common array-like things that we don't want to support, I think we'd at least need to exclude sparse *matrices* (always).

IIUC eventually sparse arrays will have an array API compliant interfaces. That would mean they *could* be added into the `array_api_compliant` test matrix and supported properly. I suppose there is still a question of whether they *should* be allowed if the algorithm does not take advantage of the sparsity.

Can I open an issue about this instead? I think it is relevant to a lot of other functions (e.g. `stats` functions never used `_asarray_validated`, but I think all the same reasoning would apply.

Reply:
> IIUC eventually sparse arrays will have an array API compliant interfaces

Not necessarily. My understanding is "let's try as far as possible, but some things might not make sense", for `scipy.sparse` (gh-18915). I think `pydata/sparse` is planning to have a compliant inferface though.

> Can I open an issue about this instead? I think it is relevant to a lot of other functions (e.g. stats functions never used _asarray_validated, but I think all the same reasoning would apply.

Sure, I was just wanting to avoid a regression, where sparse matrices were rejected but now pass through. That doesn't seem too problematic though, so as long as it is on your radar 👍.

An issue already exists, gh-18972, and I commented about this topic there last year.

Reply:
> sparse matrices were rejected but now pass through

They're still rejected, just with a little less obvious error.

```python3
from scipy import sparse, special
x = sparse.csc_array([[1, 2, 3, 0]])
special.logsumexp(x, axis=-1)
# TypeError: ufunc 'isfinite' not supported for the input types...
```

Yeah, this will be on my radar, and I see it's already tracked. Thanks!